### DISCUSSION TITLE
#include errors detected. Consider updating your compile_commands.json or includePath. Squiggles are disabled for this translation unit

### DISCUSSION THREAD

**Discussion by Benedito821** [microsoft/vscode-cpptools#7205] [Q&A]
Hello! I am facing problems when trying to build my project(I use ubuntu 18). I see these errors:

1) #include errors detected. Consider updating your compile_commands.json or includePath. Squiggles are disabled for this translation unit (/home/.../.../...).
2) cannot open source file "freertos/FreeRTOS.h"
3) cannot open source file "freertos/task.h"
4) cannot open source file "gpio.h"

I found here on the forum the same issue but there was not a solution.

I will show at once my **c_cpp_properties.json** file:
{
    "configurations": [
        {
            "name": "Linux",
            "includePath": [
                "${workspaceFolder}/**",
            ],
            "defines": [],
            "compilerPath": "/usr/bin/gcc",
            "cStandard": "gnu11",
            "cppStandard": "gnu++14",
            "intelliSenseMode": "linux-gcc-x64",
            "compileCommands": "${workspaceFolder}/build/bootloader/compile_commands.json"
        }
    ],
    "version": 4
}

my **settings_json** file : 
{
    "idf.port": "/dev/ttyUSB0",
    "idf.flashType": "UART",
    "C_Cpp.errorSquiggles": "Enabled"
}

I am stucked on this and cannot build my project
All the files exist on my disk. I tried  to add new paths in includePath but nothing changed. I don't know what to do.

**Comment by sean-mcmanus** [comment]
There errors you're showing are not related to building -- they're just IntelliSense errors. It sounds like your compile_commands.json doesn't have the include directories set correctly? If you can't build you need to adjust your build system to add the include directories. Then if the build system generates the compile_commands.json correctly then the IntelliSense errors should ago away. Otherwise, you could try removing the compileCommands and just adding the include directory to the includePath.

**Reply by Benedito821** [reply]
setting the paths doesn't really help.  Each time i open a file , I find new not included files. I set "C_Cpp.errorSquiggles": "Disabled" and the problems went away.

**Reply by sean-mcmanus** [reply]
If you run the C/C++: Log Diagnostics command you could see what includePaths we're using for IntelliSense and compare that with what is required for locating the headers when bulding to see what is missing.

**Reply by Benedito821** [reply]
How to type this command? Sorry for this may be stupid question, but whenever I tried this the terminal didn't recognize it

**Reply by sean-mcmanus** [reply]
The VS Code View->Command Palette

**Reply by Benedito821** [reply]
Even on the vs code terminal . I type C/C++: Log Diagnostics and it shows error

**Reply by sean-mcmanus** [reply]
What error? Are you using this command?
![image](https://user-images.githubusercontent.com/19859882/112177635-574e7380-8bb6-11eb-9b0d-75052f7e6c99.png)

**Reply by Benedito821** [reply]
I used this command. It listed all paths to files used in the project. The weird thing is that I also saw in this list paths to files that are complaining about "cannot open this file" in text editor. Now my project builds, although the errors persist.

**Reply by sean-mcmanus** [reply]
It shows the paths to files that can't be found in the "Includes" list? I'm not sure what is going wrong then.

**Reply by Benedito821** [reply]
yes. I reinstalled my extension. And now the folders are really missing. I added the path for the file FreeRTOS.h to the c_cpp_properties.json file . I did a C/C++:Log Diagnostics and the path appeared in Translation Unit Configurations . But there are still errors (cannot open source file).

**Reply by sean-mcmanus** [reply]
It seems like you need to add the path to freertos and not FreeRTOS.h.
